.github/workflows/lint.yml

```diff
@@ -0,0 +1,29 @@
+name: Lint
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - uses: bahmutov/npm-install@v1
+        with:
+          install-command: npm install
+      - run: npm run lint
```

Comment:
Does this work even without a package.json file in the repo?

Reply:
Ok. I see you added one but do we want one?

Reply:
yes; I think solhint is only available via npm/yarn so we need a package.json I believe